machetli/environments.py

```diff
@@ -83,6 +83,24 @@ def __init__(self, name, evaluator_path, batch_dir, tasks):
         self.tasks = tasks
 
 
+def _update_completed_task_status(task, exit_code):
+    if exit_code == EXIT_CODE_IMPROVING:
+        task.status = EvaluationTask.DONE_AND_IMPROVING
+    elif exit_code == EXIT_CODE_NOT_IMPROVING:
+        task.status = EvaluationTask.DONE_AND_NOT_IMPROVING
+    elif exit_code == 24: # TODO issue82 (timeouts)
```

Comment:
I changed this to #66. It is not quite the right issue description because it only covers adding a time limit to the environment but we can deal with it there.

Reply:
I also changed the 24 to `EXIT_CODE_RESOURCE_LIMIT`. I thought this could never be returned but in the documentation, we recommend this to users. I think the use case would be that someone starts a process within the evaluator but they limit the time for this subprocess themselves. This way, the evaluator won't run out of time but its subprocess might. In that case, the user can explicitly return `EXIT_CODE_RESOURCE_LIMIT` to signal to the search that they don't know if the behavior is still present.